subsys/bootloader/bootloader.c

```diff
@@ -50,7 +50,7 @@ static bool verify_firmware(u32_t address)
 
 	for (u32_t key_data_idx = 0; key_data_idx < num_public_keys;
 			key_data_idx++) {
-		if (public_key_data_read(key_data_idx, &key_data[0],
+		if (public_key_data_read(key_data_idx, (u32_t *)&key_data[0],
```

Comment:
Should `key_data` be a `u32_t` array to avoid the cast?

Reply:
If we do that, then we need patches two other places, instead of this one place.

1. The declaration must divide the `CONFIG_SB_PUBLIC_KEY_HASH_LEN` to get word size.
2. The `key_data` argument to `crypto_root_of_trust` must be casted to `(u8_t *)key_data.

IMO it is better to keep this one cast.

Reply:
Ah, I thought we could remove a cast. I'm inclined to say we should change it anyway, since casting from `(u32_t *)` to `(u8_t *)` is less scary than the other way around. Or at least make sure `key_data` is word-aligned

Reply:
As far as I understood OTP is limited to word-access so the key_data will always be read word-aligned. Should probably be documented.

Reply:
`key_data` is a stack variable in this case, isn't it?

Reply:
Yes its a stack variable, I agree with your point @oyvindronningstad , will change it to u32_t